core-web/libs/block-editor/src/lib/shared/services/dot-block-editor-init/dot-block-editor-init.service.ts

```diff
@@ -0,0 +1,57 @@
+import { inject, Injectable } from '@angular/core';
+
+import { DotAiService } from '../dot-ai/dot-ai.service';
+
+/**
+ * Service for initializing the Dot Block Editor external configuration.
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class DotBlockEditorInitService {
+    private dotAiService: DotAiService = inject(DotAiService);
+
+    private _isPluginInstalled = false;
+
+    get isPluginInstalled(): boolean {
+        return this._isPluginInstalled;
+    }
+
+    /**
+     * Initializes the Block Editor external configurations.
+     *
+     * @return {Promise<any>} A Promise that resolves when the Block Editor is initialized.
+     */
+    initializeBlockEditor(): Promise<unknown> {
+        return Promise.all([
+            this.verifyAiDotCMSPlugin()
+            // additional configs
+        ]);
+    }
+
+    /**
+     * Verifies the installation of the AiDotCMSPlugin.
+     *
+     * @returns {Promise<boolean>} A Promise that resolves to a boolean value indicating whether the plugin is installed or not.
+     *
+     * @throws {Error} If an error occurs while checking the plugin installation.
+     */
+    async verifyAiDotCMSPlugin(): Promise<boolean> {
+        if (this._isPluginInstalled) {
+            return Promise.resolve(this._isPluginInstalled);
+        }
+
+        try {
+            const response = await this.dotAiService.checkPluginInstallation().toPromise();
+            this._isPluginInstalled = response.status === 200;
+
+            return this._isPluginInstalled;
+        } catch (error) {
+            console.error('Error checking plugin installation:', error);
+
+            this._isPluginInstalled = false;
+
+            return this._isPluginInstalled;
+        }
```

Comment:
This code is kind of duplicated, maybe you can do it once in a `finally` block

Reply:
sort of, but it depends on the `this._isPluginInstalled = response.status === 200;` I need that assignation after to fill the injectToken—anything different to status 200 needs to be false.  Let me check if I can make it less visually redundant.

Reply:
Improved @zJaaal thanks!..